website_sale_hide_price/models/product_template.py

```diff
@@ -38,3 +38,21 @@ def _get_combination_info(
             }
         )
         return combination_info
+
+    def _search_render_results(self, fetch_fields, mapping, icon, limit):
+        results_data = super()._search_render_results(
+            fetch_fields, mapping, icon, limit
+        )
+        new_result = []
```

Comment:
You don't need to create a new list as you're update the original references so you can safely return `results_data`